aptos-move/aptos-vm-types/src/module_write_set.rs

```diff
@@ -98,8 +98,13 @@ impl ModuleWriteSet {
         module_storage: &'a impl ModuleStorage,
     ) -> impl Iterator<Item = PartialVMResult<WriteOpInfo>> {
         self.writes.iter_mut().map(move |(key, write)| {
+            // MODULE METERING SAFETY:
+            //   If there is a write to the module at key K, it means the module at K has been read,
+            //   and therefore it must have been loaded and metered. If module at K previously did
+            //   not exist, the publishing logic must have charged for it (and there will be no
+            //   module to load here anyway).
             let prev_size = module_storage
-                .fetch_module_size_in_bytes(write.module_address(), write.module_name())
+                .unmetered_get_module_size(write.module_address(), write.module_name())
```

Comment:
I wonder about high-level trade-off here between having explicit metered and unmetered APIs (used based on the explicit knowledge of caller context), or whether it's desirable (even ideally or eventually), to have some sort of gas meter / context that automatically figures it out, potentially after we have continued sessions. What do you think? cc @vgao1996

Reply:
We kind of have it with `Loader` traits right? All APIs take a meter, and I have change all module accesses in the move VM to go through `Loader` (see next PRs) - but Aptos VM is trickier so there raw accesses are still there...

Reply:
Right, that was the confusing part. I was mixing them up with the Loader traits.